### PR TITLE
SDK setup

### PR DESCRIPTION
**Pull request by PaulLeCam**

### CODE REVIEW

File: apps/developers/docs/learn/dice-stack.md [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# DICEstack
+
+## D for DID
+
+## I for IPLD
```
Comment (oed) [inline apps/developers/docs/learn/dice-stack.md:5]: Nice.

File: apps/developers/docusaurus.config.js [hunk 2]
```diff
@@ -0,0 +1,126 @@
+const lightCodeTheme = require('prism-react-renderer/themes/github')
+const darkCodeTheme = require('prism-react-renderer/themes/dracula')
+
```
Comment (oed) [inline apps/developers/docusaurus.config.js:3]: are we still using docusaurus?

File: apps/self.id/src/client/env.ts [hunk 3]
```diff
@@ -22,14 +25,13 @@ export async function authenticate(
   address: string
 ): Promise<SelfID> {
   const authProvider = new EthereumAuthProvider(provider, address)
-  const did = await client.authenticate(authProvider)
-  client.ceramic.did = did
+  const did = await client.authenticate(authProvider, true)
```
Comment (oed) [inline apps/self.id/src/client/env.ts:28]: What does true mean here?

File: packages/core/src/core.ts [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+  async getAccountDID(account: string): Promise<string> {
+    const link = await Caip10Link.fromAccount(this.#ceramic, account)
+    if (link.did == null) {
+      throw new Error(`No DID found for ${account}`)
+    }
+    return link.did
+  }
+
+  async toDID(accountOrDID: string): Promise<string> {
+    return isCAIP10string(accountOrDID) ? await this.getAccountDID(accountOrDID) : accountOrDID
```
Comment (oed) [inline packages/core/src/core.ts:82]: does js-ceramic support the new caip format already?

File: packages/core/src/public.ts [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+/**
+ * ```sh
+ * import { PublicID } from '@self.id/core'
+ * ```
+ */
+export class PublicID<
+  ModelTypes extends CoreModelTypes = CoreModelTypes,
+  Alias extends keyof ModelTypes['definitions'] = keyof ModelTypes['definitions']
+> {
+  #core: Core<ModelTypes, Alias>
```
Comment (oed) [inline packages/core/src/public.ts:20]: Can you remind me about that `#` does? :)

File: tsconfig.json [hunk 6]
```diff
@@ -1,19 +1,16 @@
 {
   "compilerOptions": {
-    "target": "es5",
-    "lib": ["dom", "dom.iterable", "esnext"],
-    "allowJs": true,
-    "skipLibCheck": true,
+    "module": "ES2020",
```
Comment (oed) [inline tsconfig.json:3]: Nice.

**Review by oed — COMMENTED** [review]
Looks good. Have one main question around the caip10 format.

**Comment by oed** [comment]
Also, would be good with some more context in the PR comment :)

File: apps/developers/docusaurus.config.js (hunk 2 above)
Comment (PaulLeCam) [reply apps/developers/docusaurus.config.js:3]: At first I was thinking of using it for the SDK docs so we could have live code examples, especially for React components having an interactive "connect" button for example.
For now we opted for having the Self.ID docs part of the Ceramic docs so this is less relevant, but I found having this docusaurus setup useful for integration tests.

File: apps/self.id/src/client/env.ts (hunk 3 above)
Comment (PaulLeCam) [reply apps/self.id/src/client/env.ts:28]: I attaches the did to ceramic, basically what the removed line used to do.

File: packages/core/src/core.ts (hunk 4 above)
Comment (PaulLeCam) [reply packages/core/src/core.ts:82]: I don't know, here we're still using the pre-v1 format, so it should be the same behavior as currently in IDX.

File: packages/core/src/public.ts (hunk 5 above)
Comment (PaulLeCam) [reply packages/core/src/public.ts:20]: It makes it a private field, as we target ES2020 we can use these new APIs :)

File: apps/self.id/src/client/env.ts (hunk 3 above)
Comment (oed) [reply apps/self.id/src/client/env.ts:28]: ok, why wouldn't it be attached by default?

File: packages/core/src/core.ts (hunk 4 above)
Comment (oed) [reply packages/core/src/core.ts:82]: oh, ok! I didn't actually check which version of the caip-js pkg you were using :)

**Review by oed — APPROVED** [review]

**Review by zachferland — APPROVED** [review]

**Pull request merged**